src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,286 @@
+    const entityLookupRequest: EntityLookupRequest = {
+      entityName: userprofile.ownerId,
+      parentId: parentContainerId
+    }
+    try {
+      const entity = await SynapseClient.lookupChildEntity(entityLookupRequest, token)
+
+      //if found - that's the folder id
+      result.dataFolderId = entity.id
+    }
+
+    catch (error) {
+      if (error.statusCode === 404) {
+        // if doesn't exist - create
+        const entityId = await this.createDataFolder(userprofile, token)
+        if (!entityId) {
+          throw ('no folder')
+        }
+        result.dataFolderId = entityId;
+      } else {
+        this.onError(error)
+        return Promise.reject(error);
+      }
+    }
+    finally {
+      // get the file listing
+      result.fileList = await this.getFileListingInFolder(token, result.dataFolderId)
+      this.setState({
+        userprofile,
+        dataFolderId: result.dataFolderId,
+        fileList: result.fileList || [],
+        isLoading: false
+      })
+      return result;
+    }
+  }
+  onError = (args: any) => {
+    console.log(args)
+  }
+
+
+  // creates data folder for user
+  createDataFolder = async (userprofile: UserProfile, token: string): Promise<string | undefined> => {
+
+    const newFolder: Entity = {
+      parentId: this.props.parentContainerId,
+      name: userprofile.ownerId,
+      concreteType: 'org.sagebionetworks.repo.model.Folder',
+    }
+    try {
+      const folder: Entity = await SynapseClient.createEntity(newFolder, token);
+      return folder.id
+    }
+    catch (error) {
+      this.onError(error)
+      throw (error)
+    }
+  }
+
+
+
+  getFileListingInFolder = async (token: string, targetFolderId: string): Promise<Entity[]> => {
+    const request = {
+      includeTypes: ['file'],
+      parentId: targetFolderId,
+      sortBy: 'NAME',
+      sortDirection: 'ASC'
+    }
+
+    try {
+      const data = await SynapseClient.getEntityChildren(request, token);
+      return data.page;
+    }
+    catch (error) {
+      this.onError(error)
+      return [];
+    }
+  }
+
+  deleteFile = async (token: string, entityId: string): Promise<any> => {
+
+    this.setState({
+      isLoading: true, modalContext: undefined
+    })
+
+    try {
+      await SynapseClient.deleteEntity(token, entityId);
+      const fileList = await this.getFileListingInFolder(token, this.state.dataFolderId!)
+      this.setState({
+        fileList
+      })
+    }
+    catch (error) {
+      this.onError(error)
+      return [];
+    }
+    finally {
+      this.setState({
+        isLoading: false
+      })
+    }
+  }
+
+  /* ------------------------------------------   rendering fns  ------------------------------------------------*/
+  /*  login control if user is not logged in */
+  renderLogin = (stateToken: string = ''): JSX.Element => {
+    let token: string | undefined = ''
+    const loggedInState: JSX.Element = <div className='bg-success text-center' role='alert'>
```

Comment:
Also to be removed.  Sign out functionality handled by other components (header nav bar dropdown in portals context).